tiledb/core.cc

```diff
@@ -1681,6 +1681,27 @@ py::str as_built_dump() {
   return res;
 }
 
+auto walk_callback = [](const char *path_ptr, tiledb_object_t obj,
+                        void *pyfunc) {
+  std::string my_path(path_ptr);
+  std::string my_objtype;
+  if (obj == TILEDB_GROUP) {
+    my_objtype = "group";
+  } else if (obj == TILEDB_ARRAY) {
+    my_objtype = "array";
+  } else {
+    return 0;
+  }
+  try {
+    py::function func =
+        py::reinterpret_borrow<py::function>((PyObject *)pyfunc);
+    func(my_path, my_objtype);
+  } catch (const std::exception &e) {
```

Comment:
Please add a test with a callback that throws a Python error to make sure this works correctly -- in the current code we are only catching `StopIteration`, and other Python errors should (I think) just be bubbled up by cython.

Reply:
You were right that all exceptions thrown from the callback were not propagated.
After my changes, the behavior is following the cython code which was only catching `StopIteration` (I used the pybind equivalent).